frontends/election-manager/src/screens/manual_data_import_precinct_screen.test.tsx

```diff
@@ -118,46 +122,72 @@ test('can enter data for candidate contests as expected', async () => {
+  expect(
+    screen.getByTestId('president-undervotes-input').closest('input')!.value
+  ).toBe('0');
+
   // We can not change the input to a non number
-  fireEvent.change(getByTestId('president-undervotes').closest('input')!, {
-    target: { value: 'daylight' },
-  });
-  expect(getByTestId('president-undervotes').closest('input')!.value).toBe('0');
+  userEvent.type(
+    screen.getByTestId('president-undervotes-input').closest('input')!,
+    'daylight'
+  );
+  expect(
+    screen.getByTestId('president-undervotes-input').closest('input')!.value
+  ).toBe('0');
+
   // We can change the input to an empty string
-  fireEvent.change(getByTestId('president-undervotes').closest('input')!, {
-    target: { value: '' },
-  });
-  expect(getByTestId('president-undervotes').closest('input')!.value).toBe('');
+  userEvent.type(
+    screen.getByTestId('president-undervotes-input').closest('input')!,
+    '{backspace}'
+  );
+  expect(
+    screen.getByTestId('president-undervotes-input').closest('input')!.value
+  ).toBe('');
 
   // We can change the input to a number
-  fireEvent.change(getByTestId('president-undervotes').closest('input')!, {
-    target: { value: '4' },
-  });
-  expect(getByTestId('president-undervotes').closest('input')!.value).toBe('4');
-  fireEvent.change(getByTestId('president-overvotes').closest('input')!, {
-    target: { value: '6' },
-  });
-  fireEvent.change(getByTestId('president-barchi-hallaren').closest('input')!, {
-    target: { value: '10' },
-  });
-  fireEvent.change(getByTestId('president-cramer-vuocolo').closest('input')!, {
-    target: { value: '20' },
-  });
-  fireEvent.change(getByTestId('president-court-blumhardt').closest('input')!, {
-    target: { value: '35' },
-  });
-  fireEvent.change(getByTestId('president-boone-lian').closest('input')!, {
-    target: { value: '25' },
+  userEvent.type(
+    screen.getByTestId('president-undervotes-input').closest('input')!,
+    '4'
+  );
+  expect(
+    screen.getByTestId('president-undervotes-input').closest('input')!.value
+  ).toBe('4');
+
+  userEvent.type(
+    screen.getByTestId('president-overvotes-input').closest('input')!,
+    '6'
+  );
+  userEvent.type(
+    screen.getByTestId('president-barchi-hallaren-input').closest('input')!,
+    '10'
+  );
+  userEvent.type(
+    screen.getByTestId('president-cramer-vuocolo-input').closest('input')!,
+    '20'
+  );
+  userEvent.type(
+    screen.getByTestId('president-court-blumhardt-input').closest('input')!,
+    '35'
+  );
+  userEvent.type(
+    screen.getByTestId('president-boone-lian-input').closest('input')!,
+    '25'
+  );
+
+  // Contest total ballots now should be updated
+  await waitFor(() => {
+    expect(screen.getByTestId('president-numBallots').textContent).toEqual(
+      '100'
+    );
   });
 
-  // A contest that does not allow write ins has no write in row.
-  expect(queryAllByTestId('president-write-in').length).toBe(0);
```

Comment:
why did you remove this check?

Reply:
Now, none of the tables have a write-in row by default, since you can add them. The difference would be that a contest that does not allow write-ins has no button to "Add Write-In Candidate," which I believe I cover in a new test.

It opens up a larger question - what if the user wants to just represent Write-Ins generically in their manual data and not specify names? Currently they can just add a "Generic" write-in candidate, but for NH this is a non-issue.